auto_nag/tests/test_utils.py

```diff
@@ -37,3 +41,54 @@ def test_hide_personal_info(self):
             "include_fields=_default,attachments&changed_field_to=" + \
             "RESOLVED&api_key=xyzxyzxyz&resolution=FIXED"
         assert "xyzxyzxyz" not in hide_personal_info(sample_exception)
+
+    def test_urljoin(self):
+        api_root = 'https://bugzilla.mozilla.org/bzapi/'
+        url = urljoin(
+            api_root, 'bug/%s/attachment?%s' %
+            (1233, '&changed_before=' +
+                '2010-12-26&product=Core,Firefox&changed_field=status&' +
+                'changed_after=2010-12-24&include_fields=_default,' +
+                'attachments&changed_field_to=RESOLVED&resolution=FIXED'))
+
+        expected_url = 'https://bugzilla.mozilla.org/bzapi/bug/1233/' + \
+            'attachment?&changed_before=2010-12-26&product=Core,Firefox&' + \
+            'changed_field=status&changed_after=2010-12-24&include_fields=' + \
+            '_default,attachments&changed_field_to=RESOLVED&resolution=FIXED'
+        assert url == expected_url
+
+    def test_qs(self):
+        param = {'product': ['Core,Firefox']}
+        _qs = qs(**param)
+        expected_qs = '&product=Core,Firefox'
+        assert _qs == expected_qs
+
+    def test_get_credentials(self):
+        # if already bztoolsrc is present, backup the existing '~/.bztoolsrc'
+        expand_usr = os.path.expanduser('~')
+        rcfile = os.path.expanduser('~/.bztoolsrc')
+        if os.path.exists(rcfile):
+            cmd = 'mv ' + rcfile + ' ' + expand_usr + '/.bztoolsrc_backup'
```

Comment:
@sylvestre It changed, 67th line is now using, execute_command() function, which does the following

+def execute_command(cmd):
+    """
+    Execute the given cmd
+    """
+    cmd = shlex.split(cmd)
+    p = subprocess.Popen(cmd, stdout=subprocess.PIPE, close_fds=True)
+    out, err = p.communicate()
+    return (out, err)
+